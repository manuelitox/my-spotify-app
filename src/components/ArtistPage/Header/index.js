import React from 'react'
import { Link } from 'react-router-dom'

import StyledHeader from './styles'
import SearchFormContainer from 'components/generic/SearchForm/container'
import { ReactComponent as HeadsetIcon } from 'svgs/headset.svg'

const Header = () => (
  <StyledHeader>
    <Link to="/">
      <HeadsetIcon height="45" width="55" viewBox="0 0 70 55"/>
    </Link>
    <SearchFormContainer placeholder="search another artist..." />
  </StyledHeader>
)

export default Header