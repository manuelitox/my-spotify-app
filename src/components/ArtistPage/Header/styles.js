import styled from 'styled-components'

const StyledHeader = styled.div`
  background-color: ${ props => props.theme.blueLight };
  display: grid;
  grid-template-columns: 1fr 1fr;
  padding: 10px 20px;
  @media only screen and (max-width: ${ props => props.theme.size.small }) {
    grid-template-columns: 1fr;
  }    
  & > div {
    justify-self: end;
    margin: 0;
    max-width: 420px;
    @media only screen and (max-width: ${ props => props.theme.size.small }) {
      justify-self: start;
      margin-top: 10px;
      padding: 0;
      max-width: 100%;
    }          
    & > div {
      top: 10px;
    }
    input {
      padding: 14px;
    }
  }
`

export default StyledHeader