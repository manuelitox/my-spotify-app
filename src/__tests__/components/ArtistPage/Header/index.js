import React from 'react'
import { shallow } from 'enzyme'

import Header from 'components/ArtistPage/Header/'

describe('Component | Header', () => {
  let component 
  beforeEach( () => {
    component = shallow(
      <Header />
    )
  })
  it('should render a component', () => {
    expect(component).toMatchSnapshot()
  })
})