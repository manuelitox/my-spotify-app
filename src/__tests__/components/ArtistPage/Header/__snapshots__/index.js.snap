// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | Header should render a component 1`] = `
<styled.div>
  <Link
    replace={false}
    to="/"
  >
    <ReactComponent
      height="45"
      viewBox="0 0 70 55"
      width="55"
    />
  </Link>
  <Connect(OnClickOutside(SearchFormContainer))
    placeholder="search another artist..."
  />
</styled.div>
`;
